### ISSUE TITLE
Integration of Vega into Workflow Reports

### ISSUE THREAD

**Issue by guerler** [galaxyproject/galaxy#19126]
This ticket serves to track the next phase of the integration: extending Vega’s capabilities into Workflow Reports. This enhancement aims to leverage Vega’s flexibility while maintaining a streamlined, user-friendly interface for report generation.

**Comment by jmchilton** [comment]
I was thinking about this after the last team meeting when you said you wanted to do a new batch of standard plots using a new toolkit. It would be really cool if you established like a set of Galaxy Markdown directives that didn't describe visualizations by visualization id but more like the type of plot. Like if we had a way to say "I want to generate a pie chart of this column of this dataset" generically - regardless of viz technology, - you could swap it out on the frontend with the current back practice and we could generate actual images on the backend for PDF exports.

**Comment by dannon** [comment]
@jmchilton yes, definitely -- in fact that's one of the reasons I picked vega for this, to use the same configurations on the front or back end to support python-based artifact generation.
